### PR TITLE
w5e1 by Liron Farzam

### PR DESCRIPTION
**Pull request by lironfarzam**
added w5e1

### CODE REVIEW

File: w5e1.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import os
+
+# pokemon s1e1 -> Ash late to professor Oak so can't choose pokemon...
+from datetime import datetime, timedelta
+
+MONDAYS = 1
+
+
+def pokemon():
+    pokemons = ['Bulbasaur', 'Squirtle', 'Charmander']
```
Comment (talbarfm) [inline w5e1.py]: Is this list going to change? if  not, it should be a tuple

Comment (talbarfm) [inline w5e1.py]: Also, you can add it as a const outside

File: w5e1.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import random
+import os
+
+# pokemon s1e1 -> Ash late to professor Oak so can't choose pokemon...
+from datetime import datetime, timedelta
+
+MONDAYS = 1
+
+
+def pokemon():
```
Comment (talbarfm) [inline w5e1.py:9]: where is the documentation?

File: w5e1.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    pokemons = ['Bulbasaur', 'Squirtle', 'Charmander']
+    starter_pokemon = random.choice(pokemons)
+    print(starter_pokemon)
```
Comment (talbarfm) [inline w5e1.py:16]: separate IO from logic   - to make the function reusable with different outputs

File: w5e1.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
+"w5e1" 
+import random
+import os
+
+# pokemon s1e1 -> Ash late to professor Oak so can't choose pokemon...
```
Comment (talbarfm) [inline w5e1.py]: You dont need this comment here - just give an indicative filename

File: w5e1.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+from datetime import datetime, timedelta
+
+MONDAYS = 1
```
Comment (talbarfm) [inline w5e1.py]: MONDAY*

File: w5e1.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+'''
+ This Is The Way
+ Write a function that gets a path to the folder,
+ and returns the list of all files that begin with the
+ "startsWith" letter sequence in that folder.
+'''
+
+
+def startsWith_files(folder, startsWith):
```
Comment (talbarfm) [inline w5e1.py]: start_with_files

Comment (talbarfm) [inline w5e1.py]: Nice, thats a generic function

Comment (talbarfm) [inline w5e1.py]: give a better name - its not the folder, its the folder_path

Comment (talbarfm) [inline w5e1.py]: snake case

Comment (talbarfm) [inline w5e1.py]: Documentation, type hints, return types

File: w5e1.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    list_files = os.listdir(folder)
+    list_deep_files = []
+    for file in list_files:
```
Comment (talbarfm) [inline w5e1.py]: this whole function can be a single line if you use list comprehension correctly

File: w5e1.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+I have no vinaigrette
+Write software that receives as input from the user two dates in the configuration: YYYY-MM-DD.
+The software will grill a new date that is between the two dates that the user entered as input.
+For example, for inputs 1912-06-23 and 1954-06-07, a possible output is 1939-09-03.
+Since I only go to the grocery store on Mondays and I am a heavy consumer of vinaigrette sauce,
+ if the date falls on Monday, print: "I do not have vinaigrette!"
+'''
+
+
+def IHaveNoVinaigrette():
```
Comment (talbarfm) [inline w5e1.py]: snake case and document function

File: w5e1.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+def startsWith_files(folder, startsWith):
+    list_files = os.listdir(folder)
+    list_deep_files = []
+    for file in list_files:
+        if file.startswith(startsWith):
+            list_deep_files.append(file)
+    return list_deep_files
+
+
+# -----------------------------------------------------------------------#
```
Comment (talbarfm) [inline w5e1.py]: remove this  ------------------

File: w5e1.py [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+'''
+I have no vinaigrette
+Write software that receives as input from the user two dates in the configuration: YYYY-MM-DD.
+The software will grill a new date that is between the two dates that the user entered as input.
+For example, for inputs 1912-06-23 and 1954-06-07, a possible output is 1939-09-03.
+Since I only go to the grocery store on Mondays and I am a heavy consumer of vinaigrette sauce,
+ if the date falls on Monday, print: "I do not have vinaigrette!"
+'''
```
Comment (talbarfm) [inline w5e1.py:45]: You dont have to copy the question in english

File: w5e1.py [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+
+def IHaveNoVinaigrette():
+    date1 = input("Enter first date: ")
+    while not is_valid_date(date1):
+        date1 = input("Enter first valid date: (YYYY-MM-DD.)")
+
```
Comment (talbarfm) [inline w5e1.py:64]: pick better variable names everywhere

File: w5e1.py [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    date2 = input("Enter second date: ")
+    while not is_valid_date(date2):
+        date2 = input("Enter second valid date: (YYYY-MM-DD.)")
+
+    date1, date2 = datetime.strptime(min(date1, date2), '%Y-%m-%d'), datetime.strptime(max(date1, date2), '%Y-%m-%d')
+    dayInWeek = get_day_of_week(random_date(date1, date2))
+    if dayInWeek == MONDAYS:
+        print("I do not have vinaigrette!")
```
Comment (talbarfm) [inline w5e1.py]: return these values, dont print them

File: w5e1.py [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+    while not is_valid_date(date1):
+        date1 = input("Enter first valid date: (YYYY-MM-DD.)")
+
+    date2 = input("Enter second date: ")
+    while not is_valid_date(date2):
+        date2 = input("Enter second valid date: (YYYY-MM-DD.)")
+
+    date1, date2 = datetime.strptime(min(date1, date2), '%Y-%m-%d'), datetime.strptime(max(date1, date2), '%Y-%m-%d')
+    dayInWeek = get_day_of_week(random_date(date1, date2))
+    if dayInWeek == MONDAYS:
```
Comment (talbarfm) [inline w5e1.py]: snake case everywhere

File: w5e1.py [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+def IHaveNoVinaigrette():
+    date1 = input("Enter first date: ")
+    while not is_valid_date(date1):
+        date1 = input("Enter first valid date: (YYYY-MM-DD.)")
+
+    date2 = input("Enter second date: ")
+    while not is_valid_date(date2):
+        date2 = input("Enter second valid date: (YYYY-MM-DD.)")
+
+    date1, date2 = datetime.strptime(min(date1, date2), '%Y-%m-%d'), datetime.strptime(max(date1, date2), '%Y-%m-%d')
```
Comment (talbarfm) [inline w5e1.py]: dont do that  - each variable in a separate line unless its a function that returns a tuple

File: w5e1.py (hunk 12 above)
Comment (talbarfm) [inline w5e1.py]: Return the string dont print it

File: w5e1.py [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+    dayInWeek = get_day_of_week(random_date(date1, date2))
+    if dayInWeek == MONDAYS:
+        print("I do not have vinaigrette!")
+    else:
+        print("Here is your sauce, it's 21.90 please")
+
+
+# function get date yyyy-mm-dd and retern if it is a valid date
+def is_valid_date(date):
```
Comment (talbarfm) [inline w5e1.py:73]: document

File: w5e1.py [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    try:
+        return datetime.strptime(date, '%Y-%m-%d')
```
Comment (talbarfm) [inline w5e1.py:80]: nice function!

File: w5e1.py [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+    except ValueError:
+        return False
+
+
+# function get start date and end date and return random date between them
+def random_date(start, end):
+    return start + timedelta(days=random.randint(0, (end - start).days))
```
Comment (talbarfm) [inline w5e1.py:91]: nice separation to functions!

File: w5e1.py [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+
+
+# function get date and return day of the week
+def get_day_of_week(date):
+    return date.weekday()
+
+
+# -----------------------------------------------------------------------#
+
+IHaveNoVinaigrette()
```
Comment (talbarfm) [inline w5e1.py]: https://www.freecodecamp.org/news/if-name-main-python-example/ add this

Comment (talbarfm) [reply w5e1.py]: read about what it means in the article and then make the code not execute when this file is imported

**Review by talbarfm — CHANGES_REQUESTED** [review]
Variable names, documentation, good function separation

File: w5e1.py (hunk 1 above)
Comment (lironfarzam) [reply w5e1.py]: POKEMONS = ('Bulbasaur', 'Squirtle', 'Charmander')

Comment (lironfarzam) [reply w5e1.py]: Not sure I understood what that meant. Just change to capslock?

File: w5e1.py (hunk 2 above)
Comment (lironfarzam) [reply w5e1.py:9]: I added documentation.

File: w5e1.py (hunk 3 above)
Comment (lironfarzam) [reply w5e1.py:16]: I chose to leave the print anyway because the function of the function was to try the random.

File: w5e1.py (hunk 4 above)
Comment (lironfarzam) [reply w5e1.py]: Delete the comment

File: w5e1.py (hunk 5 above)
Comment (lironfarzam) [reply w5e1.py]: opss.. you ara right.

File: w5e1.py (hunk 6 above)
Comment (lironfarzam) [reply w5e1.py]: Switched

Comment (lironfarzam) [reply w5e1.py]: 👍

Comment (lironfarzam) [reply w5e1.py:25]: def starts_with_files(folder_path_to_search_in, starts_with):

Comment (lironfarzam) [reply w5e1.py:25]: def starts_with_files(folder_path_to_search_in, starts_with):

Comment (lironfarzam) [reply w5e1.py]: def search_files_starts_with_in_folder(folder_path_to_search_in, starts_with):

Comment (lironfarzam) [reply w5e1.py]: search_files_starts_with_in_folder

Comment (lironfarzam) [reply w5e1.py]: added

File: w5e1.py (hunk 7 above)
Comment (lironfarzam) [reply w5e1.py]: list_deep_files = [file for file in list_files if file.startswith(starts_with)]

File: w5e1.py (hunk 8 above)
Comment (lironfarzam) [reply w5e1.py]: def i_Have_no_vinaigrette() -> str:
and document.

File: w5e1.py (hunk 9 above)
Comment (lironfarzam) [reply w5e1.py]: delete.
how is the correct way to separate different area?

File: w5e1.py (hunk 10 above)
Comment (lironfarzam) [reply w5e1.py:45]: its just for me...

File: w5e1.py (hunk 11 above)
Comment (lironfarzam) [reply w5e1.py:64]: first_date, second_date, day_in_week

File: w5e1.py (hunk 12 above)
Comment (lironfarzam) [reply w5e1.py]: return "I do not have vinaigrette!" if day_in_week == MONDAY else "Here is your sauce, it's 21.90 please"

File: w5e1.py (hunk 13 above)
Comment (lironfarzam) [reply w5e1.py]: day_in_week

File: w5e1.py (hunk 14 above)
Comment (lironfarzam) [reply w5e1.py]: Even if I want to work on the same variables? And the goal was not to set new ones

File: w5e1.py (hunk 12 above)
Comment (lironfarzam) [reply w5e1.py]: return "I do not have vinaigrette!" if day_in_week == MONDAY else "Here is your sauce, it's 21.90 please"

File: w5e1.py (hunk 15 above)
Comment (lironfarzam) [reply w5e1.py:73]: added

File: w5e1.py (hunk 16 above)
Comment (lironfarzam) [reply w5e1.py:80]: 👍

File: w5e1.py (hunk 17 above)
Comment (lironfarzam) [reply w5e1.py:91]: 👍

**Pull request merged**